core/connection.js

```diff
@@ -0,0 +1,26 @@
+const axios = require("axios");
+
+/**
+ * The connection module contains utilities for creating connections to a Textile node
+ */
+class Connection {
+  /**
+   * get() coerces the given options into a connection
+   */
+  static get(options) {
+    const opts = Connection.cleanOpts(options);
+
+    return axios.create({
+      baseURL: `${opts.url}:${opts.port}` // TODO this is flaky
```

Comment:
This assumes it will always have a port option, which it might not if the API is exposed via public URL or similar. Might want to conditionally include the `:` when a port is specified.

Reply:
I added a fix for this, but the cleanOpts method is still defaulting it to 40601 if it's empty, which means that one would have to pass in 80 or 443 in the above situation. I'm not sure here if you are looking to set the default to 80/empty or not. Feel free to change it around, just might need to change the usage in the README then as well.